egs/wsj/s5/steps/durmod/durmodel_train.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Copyright 2015 Hossein Hadian
+
+num_epochs=100
+minibatch_size=512
+train_history_size=10
+compute_prob_interval=5
+nnet_config=
+cmd=run.pl
+
+[ -f ./path.sh ] && . ./path.sh; # source the path.
+. parse_options.sh || exit 1;
+
+if [ $# != 1 ]; then
+   echo "Usage: $0 [options] <duration-model-dir>"
+   echo "e.g.: $0 exp/mono/durmod"
+   echo ""
+   echo "Main options (for others, see top of script file):"
+   echo "  --num-epochs <number>                       # max number of epochs for training"
+   echo "  --minibatch-size <size>                     # minibatch size"
+   echo "  --compute-prob-interval <int>               # interval for measuring accuracy"
+   echo "  --nnet-config <nnet3-conf-file>             # use this config for training"     
+   exit 1;
+fi
+
+dir=$1
+durmodel=$dir/durmodel.mdl
+mkdir -p $dir/log
+mkdir -p $dir/tmp
+
+
+if [[ ! -z $nnet_config ]]; then
+  echo "Using provided config file for nnet."
+  nnet3-init $nnet_config $dir/0.raw || exit 1;
+else
+  durmod-copy --raw=true $durmodel $dir/0.raw || exit 1;
+fi
+
+for epoch in $(seq 0 $[$num_epochs-1]); do
+  
+  echo "Epoch: "$epoch
+  curr_nnet=$dir/$[$epoch%($train_history_size+1)].raw
+  next_nnet=$dir/$[($epoch+1)%($train_history_size+1)].raw
+  
+  $cmd $dir/log/train_$epoch.log \
+       nnet3-train $curr_nnet \
```

Comment:
You do have access- everyone on the CLSP cluster does by default.
But you have to configure and compile on a machine with GPU e.g. b10 (or
any b or g machine); and you should always submit to the queue with --gpu 1
because otherwise you will crash someone else's job.   Also please don't
run 'make test' unless you are in a qlogin session e.g. qlogin -l gpu=1
-now no, for the same reason.  Also if you do use a qlogin session, always
terminate it promptly when you're done with the GPU.



On Sat, Jan 9, 2016 at 2:32 PM, Hossein Hadian <notifications@github.com>
wrote:

> In egs/wsj/s5/steps/durmod/durmodel_train.sh
> <https://github.com/kaldi-asr/kaldi/pull/414#discussion_r49268914>:
>
> > +
> > +if [[ ! -z $nnet_config ]]; then
> > +  echo "Using provided config file for nnet."
> > +  nnet3-init $nnet_config $dir/0.raw || exit 1;
> > +else
> > +  durmod-copy --raw=true $durmodel $dir/0.raw || exit 1;
> > +fi
> > +
> > +for epoch in $(seq 0 $[$num_epochs-1]); do
> > +
> > +  echo "Epoch: "$epoch
> > +  curr_nnet=$dir/$[$epoch%($train_history_size+1)].raw
> > +  next_nnet=$dir/$[($epoch+1)%($train_history_size+1)].raw
> > +
> > +  $cmd $dir/log/train_$epoch.log \
> > +       nnet3-train $curr_nnet \
>
> No actually it is not fast at all but I do not have access to GPU. I added
> an option --use-gpu similar to the one for train_tdnn.sh so that we can
> choose to use gpu or not.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/414/files#r49268914>.
>